OpenSim/Simulation/Model/MovingPathPoint.h

```diff
@@ -101,49 +77,50 @@ OpenSim_DECLARE_CONCRETE_OBJECT(MovingPathPoint, PathPoint);
     //--------------------------------------------------------------------------
 public:
     MovingPathPoint();
-    MovingPathPoint(const MovingPathPoint &aPoint);
     virtual ~MovingPathPoint();
 
-#ifndef SWIG
-    MovingPathPoint& operator=(const MovingPathPoint &aPoint);
-#endif
-    void copyData(const MovingPathPoint &aPoint);
     void updateFromXMLNode(SimTK::Xml::Element& aNode, int versionNumber) override;
-    void init(const PathPoint& aPoint) override;
 
-    const Coordinate* getXCoordinate() const { return _xCoordinate; }
-    const Coordinate* getYCoordinate() const { return _yCoordinate; }
-    const Coordinate* getZCoordinate() const { return _zCoordinate; }
-#ifndef SWIG
-    void setXCoordinate( const SimTK::State& s, Coordinate& aCoordinate);
-    void setYCoordinate( const SimTK::State& s, Coordinate& aCoordinate);
-    void setZCoordinate( const SimTK::State& s, Coordinate& aCoordinate);
+    bool hasXCoordinate() const;
+    bool hasYCoordinate() const;
+    bool hasZCoordinate() const;
+
+    const Coordinate& getXCoordinate() const;
+    const Coordinate& getYCoordinate() const;
+    const Coordinate& getZCoordinate() const;
+
+    void setXCoordinate(const Coordinate& coordinate);
+    void setYCoordinate(const Coordinate& coordinate);
+    void setZCoordinate(const Coordinate& coordinate);
 
     // Override methods from PathPoint.
     bool isActive(const SimTK::State& s) const override { return true; }
-    void connectToModelAndPath(Model& aModel, GeometryPath& aPath) 
-                                                                override;
-    void update(const SimTK::State& s) override;
-    void getVelocity(const SimTK::State& s, SimTK::Vec3& aVelocity) override;
-#endif
+
+    /** Get the local location of the MovingPathPoint in its Frame */
+    SimTK::Vec3 getLocation(const SimTK::State& s) const;
+    /** Get the local velocity of the MovingPathPoint w.r.t to and 
+        expressed in its Frame. To get the velocity of the point w.r.t.
+        and expressed in Ground, call getVelocityInGround(). */
```

Comment:
> I don't see a getVelocityInGround(). Was calcVelocityInGround() intended?

`getVelocityInGround()` is implemented by the base and handles caching the calculated value. The calculation is virtual and the responsibility of the derived Point classes. Perhaps I missed an `override`? I will check.